### ISSUE TITLE
JAX sum along first axis is slow

### ISSUE THREAD

**Issue by ricardoV94** [jax-ml/jax#25187]
### Description

We found this while working on the JAX backend of PyTensor: 

```python
import numpy as np
import jax

def np_sum(x):
    return np.sum(x, axis=0, dtype=np.float32)

def jax_sum(x):
    return jax.numpy.sum(x, axis=0)

jax_sum_jitted = jax.jit(jax_sum)

N = 128
x_np = np.random.default_rng(123).normal(size=(N, N, N)).astype(np.float32)
x_jax = jax.numpy.array(x_np)

np.testing.assert_allclose(np_sum(x_np), jax_sum(x_jax), atol=1e-4)
np.testing.assert_allclose(np_sum(x_np), jax_sum_jitted(x_jax), atol=1e-4)

%timeit np_sum(x_np)
%timeit jax.block_until_ready(jax_sum(x_jax))
%timeit jax.block_until_ready(jax_sum_jitted(x_jax))
463 μs ± 16.9 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
2.86 ms ± 74.3 μs per loop (mean ± std. dev. of 7 runs, 100 loops each)
2.77 ms ± 45.8 μs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

Running on a CPU

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.4.35
jaxlib: 0.4.35
numpy:  1.26.4
python: 3.12.7 | packaged by conda-forge | (main, Oct  4 2024, 16:05:46) [GCC 13.3.0]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='fedora', release='6.11.7-300.fc41.x86_64', version='#1 SMP PREEMPT_DYNAMIC Fri Nov  8 19:23:10 UTC 2024', machine='x86_64')
```

**Comment by ricardoV94** [comment]
@divyajot5005 I'm not a member of this repo, also I don't think this is a jax issue but an xla one

**Comment by jakevdp** [comment]
It seems this may have been fixed somewhere along the way: with JAX v0.5.2 on a Colab CPU runtime, I find JAX's performance similar to that of NumPy. Here's my test code:
```python
import numpy as np
import jax

def np_sum(x, axis=0):
    return np.sum(x, axis=axis, dtype=np.float32)

def jax_sum(x, axis=0):
    return jax.numpy.sum(x, axis=axis)

jax_sum_jitted = jax.jit(jax_sum, static_argnames=['axis'])

N = 128
x_np = np.random.default_rng(123).normal(size=(N, N, N)).astype(np.float32)
x_jax = jax.numpy.array(x_np)

np.testing.assert_allclose(np_sum(x_np), jax_sum(x_jax), atol=1e-4)
np.testing.assert_allclose(np_sum(x_np), jax_sum_jitted(x_jax), atol=1e-4)

print(f"{jax.__version__=}")
print(f"{jax.default_backend()=}")
for axis in [0, 1, 2]:
  print(f"\n{axis=}")
  %timeit np_sum(x_np, axis=axis)
  _ = jax.block_until_ready(jax_sum(x_jax, axis=axis))
  %timeit jax.block_until_ready(jax_sum(x_jax, axis=axis))
  _ = jax.block_until_ready(jax_sum_jitted(x_jax, axis=axis))
  %timeit jax.block_until_ready(jax_sum_jitted(x_jax, axis=axis))
```
Output:
```
jax.__version__='0.5.2'
jax.default_backend()='cpu'

axis=0
621 µs ± 98 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
621 µs ± 14.2 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
732 µs ± 122 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

axis=1
1.04 ms ± 181 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
594 µs ± 155 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
441 µs ± 13 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

axis=2
1.04 ms ± 195 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
793 µs ± 11.2 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
975 µs ± 301 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```
